### PR TITLE
Tavern tests

### PR DESCRIPTION
**Pull request by rsjostrand-hpe**
### Summary and Scope
Created disruptive and destructive Tavern API tests for the BSS APIs.

Switch the CT tests to use the hms-simulation-environment, the new tavern tests require lifelike hardware to function correctly.  

Updated Swagger file to reflect how BSS actually works.

### Issues and Related PRs

LIST AND CHARACTERIZE RELATIONSHIP TO JIRA ISSUES AND OTHER PULL REQUESTS. BE SURE LIST DEPENDENCIES.

* Resolves [CASM-XYZ](https://jira-pro.its.hpecorp.net:8443/browse/CASMHMS-5796)

### Testing

LIST THE ENVIRONMENTS IN WHICH THESE CHANGES WERE TESTED.

Tested on:

* Local system

Were the install/upgrade based validation checks/tests run?(goss tests/install-validation doc)
Were continuous integration tests run? Y/N   If not, Why? Yes
Was an Upgrade tested?                 Y/N   If not, Why? N/A
Was a Downgrade tested?                Y/N   If not, Why? N/A
If schema changes were part of this change, how were those handled in your upgrade/downgrade testing? N/A

WHAT WAS THE EXTENT OF TESTING PERFORMED? MANUAL VERSUS AUTOMATED TESTS (UNIT/SMOKE/OTHER)
HOW WERE CHANGES VERIFIED TO BE SUCCESSFUL?

Ran the updated CT tests locally and in Github Actions.

### Risks and Mitigations

HAS A SECURITY AUDIT BEEN RUN? (./runSnyk.sh)
ARE THERE KNOWN ISSUES WITH THESE CHANGES?
ANY OTHER SPECIAL CONSIDERATIONS?

INCLUDE THE FOLLOWING ITEMS THAT APPLY. LIST ADDITIONAL ITEMS AND PROVIDE MORE DETAILED INFORMATION AS APPROPRIATE.

Requires:

* Additional testing on bare-metal
* Compute nodes
* 3rd party software
* Broader integration testing
* Fresh install
* Platform upgrade

### CODE REVIEW

File: .github/workflows/run_ct_test.yaml [hunk 1]
```diff
@@ -2,6 +2,15 @@ name: Run CT Tests
 on: [push, pull_request, workflow_dispatch]
 jobs:
   run_ct_test:
-    uses: Cray-HPE/hms-build-image-workflows/.github/workflows/run_ct_test.yaml@v1
+    uses: Cray-HPE/hms-build-image-workflows/.github/workflows/run_ct_test_with_hms_sim.yaml@CASMHMS-5796
```
Comment (rsjostrand-hpe) [inline .github/workflows/run_ct_test.yaml:5]: Once https://github.com/Cray-HPE/hms-build-image-workflows/pull/8 merges need to change the tag ref to `v1`

**Review by nieuwsma — APPROVED** [review]

File: test/ct/api/2-disruptive/test_bootparameters.tavern.yaml [hunk 2]
```diff
@@ -0,0 +1,1441 @@
...
+              type: str
+              required: False
+            instance:
+              type: str
+              required: False
+            status:
+              type: int
+              required: False
+
+- name: Retrieve created Bootparameteters with xname specified as identifier
```
Comment (schooler-hpe) [inline test/ct/api/2-disruptive/test_bootparameters.tavern.yaml:188]: Small typo here.

File: test/ct/api/2-disruptive/test_bootparameters.tavern.yaml [hunk 3]
```diff
@@ -0,0 +1,1441 @@
...
+      - x9999c0s1b0n0
+      params: console=tty0 console=ttyS0,115200
+      kernel: s3://boot-images/00000000-0000-0000-0000-000000000001/kernel
+      initrd: s3://boot-images/00000000-0000-0000-0000-000000000001/initrd
+  response:
+    status_code: 200
+    headers:
+      BSS-Referral-Token: !anystr
+
+- name: Retrieve replaced Bootparameteters with xname specified as identifier
```
Comment (schooler-hpe) [inline test/ct/api/2-disruptive/test_bootparameters.tavern.yaml:292]: Typo.

File: test/ct/api/2-disruptive/test_bootparameters.tavern.yaml [hunk 4]
```diff
@@ -0,0 +1,1441 @@
...
+                      hostname:
+                        type: str
+                        enum:
+                        - ""
+                      fqdn:
+                        type: str
+                        enum:
+                        - ""
+
+- name: Update Bootparameteters with xname specified as identifier
```
Comment (schooler-hpe) [inline test/ct/api/2-disruptive/test_bootparameters.tavern.yaml:380]: Many more of these, going to stop leaving comments for them.

File: test/ct/api/2-disruptive/test_bootparameters.tavern.yaml [hunk 5]
```diff
@@ -0,0 +1,1441 @@
...
+                      hostname:
+                        type: str
+                        enum:
+                        - ""
+                      fqdn:
+                        type: str
+                        enum:
+                        - ""
+
+- name: Update bootparamter for only x9999c0s6b0n1
```
Comment (schooler-hpe) [inline test/ct/api/2-disruptive/test_bootparameters.tavern.yaml:767]: Different typo here.

**Review by schooler-hpe — APPROVED** [review]

**Pull request merged**